catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/mappings/CswRecordMapperFilterVisitor.java

```diff
@@ -178,10 +189,29 @@ public Object visit(PropertyIsEqualTo filter, Object extraData) {
             sourceIds.add((String) ((Literal) filter.getExpression2()).getValue());
             return null;
         }
+        AttributeType type =
+                attributeTypes.get(((PropertyName) filter.getExpression1()).getPropertyName());
+
+        LiteralExpressionImpl typedExpression = (LiteralExpressionImpl) filter.getExpression2();
+        if (type != null) {
```

Comment:
True. I don't think there's anything else other than dates that needs to be accounted for. This was the only way I could think of doing this without a ClassCastException